src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDaoTest.java

```diff
@@ -63,12 +84,49 @@ void upsertSnapshotBuilderSettingsUpdatesWhenExisting() {
   }
 
   @Test
-  void upsertSnapshotBuilderSettingsCreatesWhenNotExisting() {
-    snapshotBuilderSettingsDao.delete(dataset.getId());
+  void upsertSettingsBySnapshotIdUpdatesWhenExisting() {
+    assertThat(
+        "Snapshot builder settings should be the new upserted value",
+        snapshotBuilderSettingsDao.upsertSnapshotBuilderSettingsBySnapshotId(
+            snapshot.getId(), SnapshotBuilderTestData.SETTINGS),
```

Comment:
One feature of `upsert` is that it can replace the existing data, would it be possible to test that functionality here?